src/devices/Arduino/FirmataCommandSequence.cs

```diff
@@ -11,26 +11,71 @@ namespace Iot.Device.Arduino
     /// A firmata command sequence
     /// Intended to be changed to public visibility later
     /// </summary>
-    internal class FirmataCommandSequence
+    public class FirmataCommandSequence
     {
+        /// <summary>
+        /// Start of sysex command byte. Used as start byte for almost all extended commands.
+        /// </summary>
+        public const byte StartSysex = (byte)FirmataCommand.START_SYSEX;
+
+        /// <summary>
+        /// End of sysex command byte. Must end all sysex commands.
+        /// </summary>
+        public const byte EndSysex = (byte)FirmataCommand.END_SYSEX;
+
         private List<byte> _sequence;
 
         /// <summary>
         /// Create a new command sequence
         /// </summary>
         /// <param name="command">The first byte of the command</param>
-        public FirmataCommandSequence(FirmataCommand command = FirmataCommand.START_SYSEX)
+        internal FirmataCommandSequence(FirmataCommand command = FirmataCommand.START_SYSEX)
         {
             _sequence = new List<byte>()
             {
                 (byte)command
             };
         }
 
-        internal List<byte> Sequence => _sequence;
+        /// <summary>
+        /// Create a new sysex command sequence. The <see cref="StartSysex"/> byte is added automatically.
+        /// </summary>
+        public FirmataCommandSequence()
+            : this(FirmataCommand.START_SYSEX)
+        {
+        }
+
+        /// <summary>
+        /// The current sequence
+        /// </summary>
+        public IReadOnlyList<byte> Sequence => _sequence;
 
+        /// <summary>
+        /// The current length of the sequence
+        /// </summary>
         public int Length => _sequence.Count;
 
+        /// <summary>
+        /// Decode an uint from packed 7-bit data.
+        /// This way of encoding uints is only used in extension modules.
+        /// </summary>
+        /// <param name="data">Data. 5 bytes expected</param>
+        /// <param name="fromOffset">Start offset in data</param>
+        /// <returns></returns>
+        public static UInt32 DecodeUInt32(ReadOnlySpan<byte> data, int fromOffset)
+        {
+            Int32 value = data[fromOffset];
```

Comment:
you should mask that one unused bit even if it should be zero, perhaps Debug.Assert it is actually zero